src/Illuminate/Validation/Validator.php

```diff
@@ -872,7 +872,12 @@ protected function hasNotFailedPreviousRuleIfPresenceRule($rule, $attribute)
      */
     protected function validateUsingCustomRule($attribute, $value, $rule)
     {
-        $attribute = $this->replacePlaceholderInString($attribute);
+        $originalAttribute = $this->replacePlaceholderInString($attribute);
+
+        $attribute = match (true) {
+            $rule instanceof Rules\File => $attribute,
+            default => $originalAttribute,
+        };
```

Comment:
match statements with only two cases kinda defeat the purpose, I'd say
```suggestion
        $originalAttribute = $this->replacePlaceholderInString($attribute);

        $attribute = $rule instanceof Rules\File
            ? $attribute
            : $this->replacePlaceholderInString($attribute);
```